spark-connector/spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/iceberg/GravitinoIcebergCatalog.java

```diff
@@ -85,4 +92,35 @@ public Identifier[] listFunctions(String[] namespace) throws NoSuchNamespaceExce
   public UnboundFunction loadFunction(Identifier ident) throws NoSuchFunctionException {
     return ((SparkCatalog) sparkCatalog).loadFunction(ident);
   }
+
+  /**
+   * 1. The GravitinoIcebergCatalog does not extend the SparkCatalog, and it also cannot extend the
+   * SparkCatalog because the initialize method of the SparkCatalog is final and cannot be
+   * overrided. 2. If directly using `((SparkCatalog) sparkCatalog).loadProcedure()`, it will cause
+   * a Preconditions check failed, please referred to
+   * https://github.com/apache/iceberg/blob/ea916c1700ef37377d3e8bba1197b3fe32d1a248/spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/procedures/BaseProcedure.java#L121
+   */
+  @Override
+  public Procedure loadProcedure(Identifier identifier) throws NoSuchProcedureException {
+    String[] namespace = identifier.namespace();
+    String name = identifier.name();
+
+    if (isSystemNamespace(namespace)) {
+      SparkProcedures.ProcedureBuilder builder = SparkProcedures.newBuilder(name);
+      if (builder != null) {
+        return builder.withTableCatalog(this).build();
+      }
+    }
+
+    throw new NoSuchProcedureException(identifier);
+  }
+
+  @Override
+  public Catalog icebergCatalog() {
+    return ((SparkCatalog) sparkCatalog).icebergCatalog();
+  }
+
+  private static boolean isSystemNamespace(String[] namespace) {
```

Comment:
could you use reflect methods to implement?

Reply:
> could you use reflect methods to implement?

what would you want to reflect?

Reply:
why need reflect here?

Reply:
Compared with the current copy code way, using the origin method by reflection is more maintainable.

Reply:
> Compared with the current copy code way, using the origin method by reflection is more maintainable.

for `isSystemNamespace` or `loadProcedure`?

Reply:
`isSystemNamespace`

Reply:
> `isSystemNamespace`

ok.

Reply:
updated.